src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ContextAdapterBase.cs

```diff
@@ -4,25 +4,37 @@
 using Microsoft.Testing.Extensions.VSTestBridge.CommandLine;
 using Microsoft.Testing.Platform;
 using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Extensions.Messages;
+using Microsoft.Testing.Platform.Requests;
 using Microsoft.VisualStudio.TestPlatform.ObjectModel;
 using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
 
 namespace Microsoft.Testing.Extensions.VSTestBridge.ObjectModel;
 
 internal abstract class ContextAdapterBase
 {
-    protected ContextAdapterBase(ICommandLineOptions commandLineOptions)
+    protected ContextAdapterBase(ICommandLineOptions commandLineOptions, IRunSettings runSettings, ITestExecutionFilter filter)
     {
+        RunSettings = runSettings;
+
+        RoslynDebug.Assert(runSettings.SettingsXml is not null);
+
+        string? filterFromRunsettings = XElement.Parse(runSettings.SettingsXml).Descendants("TestCaseFilter").SingleOrDefault()?.Value;
```

Comment:
I am usually not a big fan of `Single` API as it throws an unhelpful exception. Ideally I'd prefer to throw ourselves an exception saying the `TestCaseFilter` appears more than once.

Reply:
I was replicating https://github.com/microsoft/testfx/blob/ab9e1494a285006703cc3703be7f4ed4268cff23/src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/RunContextAdapter.cs#L25

The logic is inaccurate anyways. We shouldn't be using `Descendants` altogether. Instead, we should look at direct child called `RunConfiguration`, then from there find `TestCaseFilter`.

Reply:
Fixed.